internal/cluster/cluster.go

```diff
@@ -701,6 +707,23 @@ func CanDownloadKubeconfig(c *common.Cluster) (err error) {
 	return err
 }
 
+func CanDownloadKubeconfigNoIngress(c *common.Cluster, agentInstaller bool) (err error) {
```

Comment:
I'm not sure about the benefits of pushing down agent specific logic at this level, and it looks like some duplication has been added.  Probably by refactoring

```
if !funk.Contains(allowedStatuses, clusterStatus) {
		err = errors.Errorf("cluster %s is in %s state, %s can be downloaded only when status is one of: %s",
			c.ID, clusterStatus, constants.Kubeconfig, allowedStatuses)
	}
```
may help in addressing it